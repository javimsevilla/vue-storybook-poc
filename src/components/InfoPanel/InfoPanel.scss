.info__panel {
  @apply rounded-b-lg shadow-md;

  display: flex;
  font-family: 'Manrope', sans-serif;
  padding: 16px;
  width: 500px;

  i {
    font-size: 16px;
    margin-right: 10px;
  }

  &__description,
  &__title {
    margin: 0;
  }

  &__title {
    font-weight: bold;
    margin-bottom: 4px;
  }

  &--info {
    background-color: #c4c4c4;

    i {
      color: gray;
    }
  }

  &--warning {
    background-color: rgba(255, 255, 113, 0.54);

    i {
      color: rgb(255, 217, 0);
    }
  }

  &--error {
    background-color: #ffb7b7;

    i {
      color: red;
    }
  }
}
