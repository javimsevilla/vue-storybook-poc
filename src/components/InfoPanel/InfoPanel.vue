<template>
  <div :class="classes">
    <i v-if="icon" :class="icon"></i>
    <div class="info__panel__text">
      <p v-if="title" class="info__panel__title">{{ title }}</p>
      <p class="info__panel__description">{{ description }}</p>
    </div>
  </div>
</template>

<style lang="scss" src="./InfoPanel.scss"></style>

<script>
import { reactive, computed } from 'vue';

export default {
  name: 'info-panel',

  props: {
    description: {
      type: String,
      required: true
    },
    title: {
      type: String
    },
    icon: {
      type: Boolean,
      default: false,
    },
    type: {
      type: String,
      validator: function (value) {
        return ['info', 'warning', 'error'].includes(value);
      },
    }
  },

  setup(props) {
    props = reactive(props);

    return {
      classes: computed(() => ({
        'info__panel': true,
        [`info__panel--${props.type}`]: true
      })),
      icon: computed(() => {
        if (!props.icon) {
          return;
        }

        const icon = {
          info: 'info',
          warning: 'warning',
          error: 'warning'
        }[props.type];

        return `icon-${icon}`;
      }),
      description: computed(() => props.description),
      title: computed(() => props.title)
    }
  },
};
</script>
