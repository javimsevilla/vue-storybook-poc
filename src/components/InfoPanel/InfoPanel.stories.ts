// InfoPanel.stories.ts

import InfoPanel from './InfoPanel.vue';

import { Meta, StoryFn } from '@storybook/vue3';

export default {
  /* üëá The title prop is optional.
   * See https://storybook.js.org/docs/vue/configure/overview#configure-story-loading
   * to learn how to generate automatic titles
   */
  title: 'Panels/Info Panel',
  component: InfoPanel,
  argTypes: {
    type: {
      control: { type: 'select' },
      options: ['info', 'warning', 'error'],
    },
  }
} as Meta<typeof InfoPanel>;

//üëá We create a ‚Äútemplate‚Äù of how args map to rendering
const Template: StoryFn<typeof InfoPanel> = (args) => ({
  components: { InfoPanel },
  setup() {
    return { args };
  },
  template: '<InfoPanel v-bind="args" />',
});

const description = `
  Para obtener una salud √≥ptima, resuelve las siguientes acciones.
  Esto te ayudar√° a mejorar tu salud contable y que el dashboard
  del trust sea representativo.
`;

export const Info = Template.bind({});

Info.args = {
  description,
  title: 'Title',
  icon: true,
  type: 'info'
};

export const Warning = Template.bind({});

Warning.args = {
  description,
  title: 'Title',
  icon: true,
  type: 'warning'
};

export const Error = Template.bind({});

Error.args = {
  description,
  title: 'Title',
  icon: true,
  type: 'error'
};
