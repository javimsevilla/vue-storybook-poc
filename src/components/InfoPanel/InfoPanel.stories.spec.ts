import { mount } from '@vue/test-utils';
import InfoPanel from './InfoPanel.vue';
import { Info } from './InfoPanel.stories';

it('Renders expected type classname', () => {
  const wrapper = mount(InfoPanel, {
    propsData: Info.args,
  });

  expect(wrapper.find('.info__panel').classes()).toContain(`info__panel--${(Info.args as any).type}`);
});

it('renders snapshot', () => {
  const wrapper = mount(InfoPanel, {
    propsData: Info.args,
  });

  expect(wrapper.element).toMatchSnapshot();
});
