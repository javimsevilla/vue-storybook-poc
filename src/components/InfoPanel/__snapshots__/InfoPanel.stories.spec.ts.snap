// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders snapshot 1`] = `
<div
  class="info__panel info__panel--info"
>
  <i
    class="icon-info"
  />
  <div
    class="info__panel__text"
  >
    <p
      class="info__panel__title"
    >
      Title
    </p>
    <p
      class="info__panel__description"
    >
      
  Para obtener una salud óptima, resuelve las siguientes acciones.
  Esto te ayudará a mejorar tu salud contable y que el dashboard
  del trust sea representativo.

    </p>
  </div>
</div>
`;
